Modules/SubModule_Import/xASL_imp_DetermineStructureFromSourcedata.m

```diff
@@ -0,0 +1,461 @@
+        end
+    end
+
+end
+function xASL_imp_PrintRun(x,thisSubject,thisVisit,thisRun)
+
+    % Print each individual run (= session in legacy terminology)
+    if isfield(x.overview.(thisSubject).(thisVisit).(thisRun),'name')
+        run = x.overview.(thisSubject).(thisVisit).(thisRun).name;
+    else
+        run = '';
+    end
+    fprintf('Run:     %s\n', run);
+
+end
+
+
+
+%% Add subjects to overview
+function x = xASL_imp_AddSubjectOverview(x)
+    
+    for iSubject=1:numel(x.modules.import.listsIDs.subjectIDs)
+        thisSubject = x.modules.import.listsIDs.subjectIDs{iSubject};
+        x = xASL_imp_AddSubject(x,thisSubject,iSubject);
+    end
+
+end
+
+
+%% Add single subject to overview
+function x = xASL_imp_AddSubject(x,thisSubject,iSubject)
+
+    % Add subject name
+    subjectFieldName = ['subject_' num2str(iSubject,'%03.f')];
+    x.overview.(subjectFieldName).name = thisSubject;
+
+    % Get vSubjectIDs
+    vSubjectIDs = strcmp(x.modules.import.listsIDs.vSubjectIDs,thisSubject);
+    
+    % Get visits of current subject
+    currentVisitList = unique(x.modules.import.listsIDs.vVisitIDs(vSubjectIDs));
+    
+    % Determine visitIDs
+    x.overview.(subjectFieldName).visitIDs = unique(x.modules.import.listsIDs.vVisitIDs(vSubjectIDs));
+    
+    for iVisit=1:numel(currentVisitList)
+        thisVisit = currentVisitList{iVisit};
+        x = xASL_imp_AddVisit(x,subjectFieldName,vSubjectIDs,thisVisit,iVisit);
+    end
+    
+
+end
+
+
+
+%% Add single visit to overview
+function x = xASL_imp_AddVisit(x,sFieldName,vSubjectIDs,thisVisit,iVisit)
+
+
+    %% Add basic visit fields
+
+    % Determine visit field name
+    vFieldName = ['visit_' num2str(iVisit,'%03.f')];
+    
+    % Get vVisitIDs
+    vVisitIDs = strcmp(x.modules.import.listsIDs.vVisitIDs,thisVisit);
+    vVisitIDs = vSubjectIDs & vVisitIDs;
+    
+    % Assign visit name and sessions
+    x.overview.(sFieldName).(vFieldName).name = thisVisit;
+    x.overview.(sFieldName).(vFieldName).sessions = unique(x.modules.import.listsIDs.vSessionIDs(vVisitIDs));
+    
+    % Get number of visits
+    x.overview.(sFieldName).nVisits = length(x.overview.(sFieldName).visitIDs);
+    
+    % Sort by output
+    if length(x.overview.(sFieldName).visitIDs)>1
+        for iV=1:numel(x.overview.(sFieldName).visitIDs)
+            IDrow(iV) = find(cellfun(@(y) strcmp(y,x.overview.(sFieldName).visitIDs{iV}), x.modules.import.imPar.tokenVisitAliases(:,1)));
+        end
+        x.overview.(sFieldName).listsIDs.visitIDs = x.overview.(sFieldName).visitIDs(IDrow);
+    end
+    
+    % Add additional fields
+    x = xASL_imp_thisSubjectVisit(x,sFieldName,vVisitIDs,vFieldName);
+    
+    
+    %% Sanity check for missing elements
+    xASL_imp_DCM2NII_SanityChecks(x,x.overview.(sFieldName),x.overview.(sFieldName).(vFieldName));
+    
+    
+    %% Preallocate space for (global) counts
+    [x.overview.(sFieldName),x.overview.(sFieldName).(vFieldName)] = ...
```

Comment:
Because I have to rename things a million scripts in that case... And please don't say it's just a CTRL F thing, because it always lead to more bugs in the past...